src/mpid/ch4/shm/xpmem/xpmem_pre.h

```diff
@@ -14,6 +14,7 @@
 #define MPIDI_XPMEM_SEG_PREALLOC 8      /* Number of segments to preallocate in the "direct" block */
 
 typedef struct MPIDI_XPMEM_seg {
+    MPIR_OBJECT_HEADER;
     /* AVL-tree internal components start */
```

Comment:
After rebased over #4141 , you should only add `MPIR_OBJECT_HEADER` in  `MPIDI_XPMEM_seg` in the "fix previous obj kind bug" commit.